### PR TITLE
feat(docs): new docs overview page + platform overview pages with configurators

### PR DESCRIPTION
**Pull request by timmywil**
### PR Checklist

Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] I am requesting to **pull a topic/feature/bugfix branch** (right side). In other words, not _develop_.
- [x] I have run `npm test` against my changes and tests pass.
- [x] I have added or edited necessary documentation, or no docs changes are needed.

### Description

Adds all overview pages to docs, including a global overview page and then one for each platform.

Each overview page has a configurator on it. The configurator was added by switching all markdown parsing from remark to [mdx](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/). This allows us to use jsx-like syntax in markdown and include React components in markdown files.

The nav was also redesigned for the whole docs section, and more related resources were added to all docs pages.

This PR does not include:

- Spotlight testing in the configurator. This will come in a separate PR. I'll do this before adding Pro now that the public API is available.
- Related videos. I'm not sure how to add these. We don't maintain a list of videos with tags on each video. This would likely involve adding lists of videos on each docs page statically, which would be error-prone and just about impossible to maintain. But, I think embedding the videos that pertain to a particular page's subject will be enough and this section isn't really needed anyway.
- There was a design for a "Quick Start" guide for Android, but we already have "Getting Started" guides for all platforms, which seems like the same thing so I left that alone. If the content should be changed on any of those pages, I leave that to others.

Remaining items:

- [x] Content reviews. I've scaffolded out all the overview pages with what was in the design for the Android page, but I'm sure you all will have ideas about what could go on each platform overview page. To review the new content, use the "Jump to" function to go to `overview.mdx` files. Also please review code samples (in the `samples` folder), especially new NLU and ASR samples.
- [x] Add code samples for NLU and ASR to the configurator
- [x] Cross-browser testing

### CODE REVIEW

**Comment by space-pope** [comment]
I like all the new stuff. Here's what I noticed on first look:

- The markdown rules (`---`) rendering as hyphens looks a bit odd
- I like the changes to the lefthand nav, though the indent is subtle enough that it took me a few looks to see it
- It'd be nice if the configurator were a generic component that we could configure inline rather than a component with a hardcoded set of features to display. i.e., it might be helpful to have an inline configurator that switches between programming languages for a given sample or one that includes a feature not available on all platforms.

I'll add suggestions for the Android stuff in a PR on top of this one.

**Comment by space-pope** [comment]
Oh, and looks like the configurator can only display one-digit line numbers (they go from `9` to `0`).

**Comment by timmywil** [comment]
> The markdown rules (---) rendering as hyphens looks a bit odd

I'm not sure what you mean by this. It renders `<hr />` elements.

> it might be helpful to have an inline configurator that switches between programming languages

That sounds different. The configurator is there to showcase platform code in the docs and on feature pages. Sample code can still be prettified on its own. That said, we could have another sample code component that switches between different languages (like Java and Kotlin) on a platform page. But putting the code for this configurator inline would be an awful mess.

> one that includes a feature not available on all platforms

You can set both platform and feature props to limit it to showing just one feature on one platform. It wouldn't be hard to limit a particular feature to 2-3 platforms, which we've done before when node and react native didn't have a speech pipeline yet. But, ideally, the different platforms have feature parity.

**Comment by space-pope** [comment]
image
Guess those are too long to be actual hyphens, but that's how the markdown `---` in `Android/overview.mdx` is rendering for me locally.

And I didn't mean putting the entire configurator inline, but maybe being able to feed the component a list of title/sample code for ad hoc display. A separate sample code component seems like a reasonable middle ground, but I don't have a concrete use case for it in mind yet, so don't think there's any need to actually add one yet.

**Comment by timmywil** [comment]
ah, that's the new style for `<hr>` elements.

**Comment by space-pope** [comment]
Gotcha. Just seems a little choppy to me. I think another thing that grabs my eye (which might be a significant part of the "choppiness") is that the `hr`/box border color is the same as the text color in dark mode (which is what I use), but it's more muted relative to the text in light mode. Makes the light mode look a little cleaner IMO.

**Edit**: Oh! I think the foreground thing also applies to the section names/individual doc page titles on the lefthand nav (I think they're the same color in dark mode). This explains so much for me...

**Comment by timmywil** [comment]
Good points. I can update both.

**Comment by timmywil** [comment]
Everything mentioned should be updated now.

**Comment by space-pope** [comment]
Awesome! I think the `hr`s could be a _little_ lighter, but that's kind of nitpicking—I like these changes a lot.

It does look like the footer's CSS is coming from somewhere other than the place that changed, though—its `hr`s are still white (almost looks like the box borders are too, but turns out they're not, just thin enough that they almost blend in with the rule).

**Comment by timmywil** [comment]
I don't know of an `hr` in the footer.

**Comment by timmywil** [comment]
Oh, you mean in the Related section. Yea, those aren't `hr`'s exactly, but should be lighter.

**Comment by space-pope** [comment]
Oh, my bad. Forgot those were inline components—yeah, that section.

File: src/components/Configurator.tsx [hunk 1]
```diff
@@ -9,165 +9,173 @@ import * as rn from '../samples/rn'
...
+    samples: python
+  }
+}
+
 export interface ConfiguratorProps {
-  codeKey: 'wakeword' | 'keyword' | 'tts'
+  feature?: Feature
+  hideLinks?: boolean
   extraCss?: SerializedStyles | SerializedStyles[]
+  platform?: Platform
```
Comment (noelweichbrodt) [inline src/components/Configurator.tsx:59]: if `platform` is specificed, then "Select Feature" should not be shown imo.

**Review by noelweichbrodt — COMMENTED** [review]
Looks great! Love the new details like the right-side outline and the redesigned Related footers.

File: src/components/Configurator.tsx (hunk 1 above)
Comment (timmywil) [reply src/components/Configurator.tsx:59]: Yea, I considered that. The design shows the header on the platform pages, but we could make it conditional.

Comment (noelweichbrodt) [reply src/components/Configurator.tsx:59]: I think that would be good. It feels redundant on the platform pages, the place we want the eye to fall is on the feature names directly

**Review by will-rice — APPROVED** [review]

File: src/samples/python.ts [hunk 2]
```diff
@@ -29,3 +29,26 @@ output = PyAudioOutput()
 manager = TextToSpeechManager(client, output)
 manager.synthesize("welcome to spokestack")
 `
+
+export const nlu = `from spokestack.nlu.tflite import TFLiteNLU
+nlu = TFLiteNLU("model_dir")
+
+## ...
```
Comment (will-rice) [inline src/samples/python.ts]: we could probably remove `## ...`

**Review by will-rice — CHANGES_REQUESTED** [review]
I tagged one fix, everything else in Python looks good.

**Review by will-rice — APPROVED** [review]
lgtm

**Pull request merged**